ESP32-S2: Building samples/hello_world for esp32s2_saola fails with CONFIG_BOOTLOADER_MCUBOOT=y

**Describe the bug**
When building the `samples/hello_world` app at [commit 792469aae95b01736495cd41ac39a64568cf9028](https://github.com/zephyrproject-rtos/zephyr/commit/792469aae95b01736495cd41ac39a64568cf9028) for MCUBoot for the `esp32s2_saola` board, the build fails.

**To Reproduce**
Steps to reproduce the behavior:
1. Set up python virtual environment

   ```bash
   cd ${HOME}
   mkdir -p ${HOME}/zephyrproject
   python3 -m venv ${HOME}/zephyrproject/.venv
   ```

2. Get Zephyr and install python deps
   ```bash
   source ${HOME}/zephyrproject/.venv/bin/activate
   pip install west
   west init ${HOME}/zephyrproject
   cd ${HOME}/zephyrproject
   west update
   west zephyr-export
   pip install -r ${HOME}/zephyrproject/zephyr/scripts/requirements.txt
   pip install kconfiglib
   ```

3. Install Zephyr SDK
   ```bash
   wget https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.15.2/zephyr-sdk-0.15.2_linux-x86_64.tar.gz
   wget -O - https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.15.2/sha256.sum | shasum --check --ignore-missing
   mkdir zephyr-sdk
   tar xf zephyr-sdk-0.15.2_linux-x86_64.tar.gz --strip=1 --directory zephyr-sdk
   rm zephyr-sdk-0.15.2_linux-x86_64.tar.gz
   cd zephyr-sdk
   # Install esp32s2 toolchain and host tools; register cmake package
   echo | ./setup.sh -t xtensa-espressif_esp32s2_zephyr-elf -h -c
   ```

4. Build `hello_world` example for `esp32s2_saola`
   ```bash
   cd ${HOME}/zephyrproject/zephyr
   # Set up Kconfig for MCUBoot
   echo "CONFIG_BOOTLOADER_MCUBOOT=y" >> samples/hello_world/prj.conf
   echo "CONFIG_BOOTLOADER_ESP_IDF=n" >> samples/hello_world/prj.conf
   west build -p always -b esp32s2_saola samples/hello_world
   ```

5. See error in Section "Logs and console output" below

**Expected behavior**

The build command `west build -p always -b esp32s2_saola samples/hello_world` succeeds.

Note that building from the same source for the `esp32` board completes with no error.

```bash
(.venv) esp@ea794260ae1d:~/zephyrproject/zephyr$ west build -p always -b esp32 samples/hello_world
[169/169] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
     mcuboot_hdr:          32 B         32 B    100.00%
        metadata:          32 B         32 B    100.00%
             ROM:       79020 B    4194240 B      1.88%
     iram0_0_seg:       11376 B        76 KB     14.62%
     irom0_0_seg:       13548 B      3264 KB      0.41%
     dram0_0_seg:        5496 B     180736 B      3.04%
     dram0_1_seg:         53 KB     110032 B     49.32%
     drom0_0_seg:        3664 B    4194240 B      0.09%
    rtc_iram_seg:          0 GB         8 KB      0.00%
    rtc_slow_seg:          0 GB         4 KB      0.00%
        IDT_LIST:          0 GB         8 KB      0.00%
```

**Impact**
Blocking Zephyr development with MCUBoot as the bootloader on ESP32-S2.

**Logs and console output**
```text
-- west build: making build dir /home/esp/zephyrproject/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/esp/zephyrproject/zephyr/samples/hello_world
-- CMake version: 3.22.1
-- Found Python3: /home/esp/zephyrproject/.venv/bin/python3.10 (found suitable exact version "3.10.6") found components: Interpreter
-- Cache files will be written to: /home/esp/.cache/zephyr
-- Zephyr version: 3.3.99 (/home/esp/zephyrproject/zephyr)
-- Found west (found suitable version "1.0.0", minimum required is "0.7.1")
-- Board: esp32s2_saola
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.15.2 (/home/esp/zephyr-sdk)
-- Found toolchain: zephyr 0.15.2 (/home/esp/zephyr-sdk)
-- Found Dtc: /home/esp/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/esp/zephyrproject/zephyr/boards/xtensa/esp32s2_saola/esp32s2_saola.dts
-- Generated zephyr.dts: /home/esp/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/esp/zephyrproject/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /home/esp/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /home/esp/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/esp/zephyrproject/zephyr/boards/xtensa/esp32s2_saola/esp32s2_saola_defconfig'
Merged configuration '/home/esp/zephyrproject/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/esp/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/esp/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/esp/zephyr-sdk/xtensa-espressif_esp32s2_zephyr-elf/bin/xtensa-espressif_esp32s2_zephyr-elf-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/esp/zephyrproject/zephyr/build
-- west build: building application
[2/168] Preparing syscall dependency handling

[3/168] Generating include/generated/version.h
-- Zephyr version: 3.3.99 (/home/esp/zephyrproject/zephyr), build: zephyr-v3.3.0-530-g1751c8f0f59d
[53/168] Building C object zephyr/CMake...hyr.dir/soc/xtensa/esp32s2/loader.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/soc/xtensa/esp32s2/loader.c.obj
ccache /home/esp/zephyr-sdk/xtensa-espressif_esp32s2_zephyr-elf/bin/xtensa-espressif_esp32s2_zephyr-elf-gcc -DCONFIG_SPI_FLASH_ROM_DRIVER_PATCH -DCONFIG_SPI_FLASH_USE_LEGACY_IMPL -DESP_PLATFORM -DKERNEL -D__ZEPHYR__=1 -I/home/esp/zephyrproject/zephyr/kernel/include -I/home/esp/zephyrproject/zephyr/arch/xtensa/include -I/home/esp/zephyrproject/zephyr/include -I/home/esp/zephyrproject/zephyr/build/zephyr/include/generated -I/home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2 -I/home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/include/crypto -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/include/wifi -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../esp_shared/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../esp_shared/include/wifi -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../esp_shared/components/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/hal/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/hal/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/hal/platform_port/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_hw_support/port/esp32s2 -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_hw_support/port/esp32s2/private_include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_hw_support/include/soc -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_hw_support/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_common/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_pm/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_rom/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_rom/esp32s2/ld -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/driver/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/soc/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/xtensa/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/xtensa/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/soc/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/soc/include/soc -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/driver/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_system/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_system/port/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_system/port/public_compat -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_wifi/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_wifi/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_phy/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_phy/include/esp32s2 -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_phy/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_timer/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_timer/private_include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_netif/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/esp_event/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/log/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/port/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/src -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/include/esp_supplicant -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/esp_supplicant/src -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/esp_supplicant/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/src/utils -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/wpa_supplicant/src/eap_peer -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/spi_flash/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/spi_flash/include/spi_flash -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/spi_flash/private_include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/bootloader_support/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/efuse/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/efuse/include/esp32s2 -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/efuse/esp32s2/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/bootloader_support/include_bootloader -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/heap/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/mbedtls/port/include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/efuse/private_include -I/home/esp/zephyrproject/modules/hal/espressif/zephyr/esp32s2/../../components/efuse/esp32s2/private_include -isystem /home/esp/zephyrproject/zephyr/lib/libc/minimal/include -fno-strict-aliasing -Os -imacros /home/esp/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always --sysroot=/home/esp/zephyr-sdk/xtensa-espressif_esp32s2_zephyr-elf/xtensa-espressif_esp32s2_zephyr-elf -imacros /home/esp/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/esp/zephyrproject/zephyr/samples/hello_world=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/esp/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/esp/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -mlongcalls -Wno-unused-variable -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/soc/xtensa/esp32s2/loader.c.obj -MF zephyr/CMakeFiles/zephyr.dir/soc/xtensa/esp32s2/loader.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/soc/xtensa/esp32s2/loader.c.obj -c /home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2/loader.c
In file included from /home/esp/zephyrproject/zephyr/include/zephyr/arch/xtensa/arch.h:18,
                 from /home/esp/zephyrproject/zephyr/include/zephyr/arch/cpu.h:27,
                 from /home/esp/zephyrproject/zephyr/include/zephyr/kernel_includes.h:33,
                 from /home/esp/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from /home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2/loader.c:8:
/home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2/loader.c: In function 'map_rom_segments':
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:192:36: error: 'DT_N_NODELABEL_image_0_REG_IDX_0_VAL_ADDRESS' undeclared (first use in this function); did you mean 'DT_N_S_cpus_S_cpu_0_REG_IDX_0_VAL_ADDRESS'?
  192 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:4043:33: note: in definition of macro 'DT_CAT4'
 4043 | #define DT_CAT4(a1, a2, a3, a4) a1 ## a2 ## a3 ## a4
      |                                 ^~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:2192:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2192 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:299:39: note: in expansion of macro 'DT_REG_ADDR'
  299 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:192:29: note: in expansion of macro 'DT_CAT'
  192 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:299:51: note: in expansion of macro 'DT_NODELABEL'
  299 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2/loader.c:33:36: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   33 |         size_t _partition_offset = FIXED_PARTITION_OFFSET(image_0);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:192:36: note: each undeclared identifier is reported only once for each function it appears in
  192 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:4043:33: note: in definition of macro 'DT_CAT4'
 4043 | #define DT_CAT4(a1, a2, a3, a4) a1 ## a2 ## a3 ## a4
      |                                 ^~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:2192:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2192 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:299:39: note: in expansion of macro 'DT_REG_ADDR'
  299 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/devicetree.h:192:29: note: in expansion of macro 'DT_CAT'
  192 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/esp/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:299:51: note: in expansion of macro 'DT_NODELABEL'
  299 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/esp/zephyrproject/zephyr/soc/xtensa/esp32s2/loader.c:33:36: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   33 |         size_t _partition_offset = FIXED_PARTITION_OFFSET(image_0);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
[58/168] Building C object zephyr/CMake...s/esp_system/port/soc/esp32s2/clk.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/esp/zephyrproject/zephyr/build
```

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 22.04 x86_64
 - Toolchain: See repro steps
 - Commit SHA
  ```text
  commit 1751c8f0f59df423fb31231856031776cf78f7ce (HEAD -> main, origin/main, origin/HEAD)
  Author: Fabio Baltieri <fabiobaltieri@google.com>
  Date:   Mon Feb 27 18:57:39 2023 +0000
  ```

**Additional context**

Building the sample hello_world app without setting `CONFIG_BOOTLOADER_MCUBOOT=y` for `eps32s2_saola` succeeds.

--- a/soc/xtensa/esp32s2/loader.c
+++ b/soc/xtensa/esp32s2/loader.c
@@ -30,7 +30,7 @@ static int map_rom_segments(void)
 {
 	int rc = 0;
 
-	size_t _partition_offset = FIXED_PARTITION_OFFSET(image_0);
+	size_t _partition_offset = FIXED_PARTITION_OFFSET(slot0_partition);
 	uint32_t _app_irom_start = _partition_offset + (uint32_t)&_image_irom_start;
 	uint32_t _app_irom_size = (uint32_t)&_image_irom_size;
 	uint32_t _app_irom_vaddr = (uint32_t)&_image_irom_vaddr;

